### ISSUE TITLE
Write sqeeze step -> ensemble step

### ISSUE THREAD

**Issue by kordc** [SebChw/Actually-Robust-Training#147]

**Comment by kordc** [comment]
I'd focus there on the 2 steps he tells us about:
* ensembles
* leave it training -> but this one may be relevant.

So maybe instead of the squeeze, we should enable ensemble DL training?

**Comment by kordc** [comment]
We agreed on making just ensemble step

**Comment by kordc** [comment]
Deep learning ensembles' training is a huge topic. I found a nice package [Ensembles-PyTorch](https://ensemble-pytorch.readthedocs.io/en/latest/quick_start.html). They provide a wide selection of already implemented ensemble types. I'll play around with this library using our ArtModule

**Comment by SebChw** [comment]
Nice, if we won't need to write it by ourselves it would be great.

**Comment by kordc** [comment]
It supports only classification and regression. Even multi-label classification is not supported.

I think doing this step a task-agnostic is simply not possible. I see three options:
* Do separate steps for classification, regression, multi-label classification etc.
* Do one step, that will create a very simple ensemble basing on voting, especially using torch.mean(). This could be calculated by for at least regression, and both types of classification. For other it's simply not that simple
* Do a template step as we do with DataAnalisys

**Comment by trebacz626** [comment]
I think weighted mean might be good enough

**Comment by trebacz626** [comment]
user can specify weights

**Comment by mmaecki** [comment]
Same as @trebacz626 suggests. Any more sophisticated ensembles can be done by the user.

**Comment by SebChw** [comment]
Actually, most tasks are classification or regression. If we can utilize this library quickly/easily lets use it, even though it doesn't support everything. If user wants something more sophisticated they must implement it by themselves.
